packages/python/plotly/plotly/express/_core.py

```diff
@@ -1599,6 +1603,31 @@ def aggfunc_continuous(x):
     return args
 
 
+def process_dataframe_timeline(args):
+    """
+    Massage input for bar traces for px.timeline()
+    """
+    args["is_timeline"] = True
+    if args["x_start"] is None or args["x_end"] is None:
+        raise ValueError("Both x_start and x_end are required")
+
+    try:
+        x_start = pd.to_datetime(args["data_frame"][args["x_start"]])
+        x_end = pd.to_datetime(args["data_frame"][args["x_end"]])
+    except (ValueError, TypeError):
+        raise TypeError(
+            "Both x_start and x_end must refer to data convertible to datetimes."
+        )
+
+    # note that we are not adding any columns to the data frame here, so no risk of overwrite
```

Comment:
not sure I understand here; there's no risk of overwrite because the dataframe is copied?

Reply:
This is saying that we're not adding a new column to the DF, we're just overwriting an existing known key that we control. If we were to add a new column to the DF called `x` or `base` we would run the risk of clobbering an existing column that was mapped to something like `color` and things would break.

Reply:
Although... as I write this, I realized that actually this exact situation breaks in this case: if someone maps `color="x_end"` we will end up clobbering their column... same problem as we encounter in sunburst/treemap when we mess with the DF :)

Reply:
thanks for the explanations. It's indeed a similar problem as the sunburst / treemap case.